K2Bridge/Factories/BucketFactory.cs

```diff
@@ -24,15 +24,30 @@ public static DateHistogramBucket CreateFromDataRow(DataRow row)
             Ensure.IsNotNull(row, nameof(row));
 
             var timestamp = row[(int)DateHistogramBucketColumnNames.Timestamp];
-            var count = row[(int)DateHistogramBucketColumnNames.Count];
+            var count = row[DateHistogramBucketColumnNames.Count];
             var dateBucket = (DateTime)timestamp;
 
-            return new DateHistogramBucket
+            var dhb = new DateHistogramBucket
             {
                 DocCount = Convert.ToInt32(count),
                 Key = TimeUtils.ToEpochMilliseconds(dateBucket),
                 KeyAsString = dateBucket.ToString("yyyy-MM-ddTHH:mm:ss.fffK"),
+                Aggs = new System.Collections.Generic.Dictionary<string, System.Collections.Generic.List<double>>(),
             };
+
+            var clmns = row.Table.Columns;
+            foreach (DataColumn clmn in clmns)
+            {
+                // todo: fix
+                if (clmn.ColumnName == DateHistogramBucketColumnNames.Count || clmns.IndexOf(clmn) == (int)DateHistogramBucketColumnNames.Timestamp)
```

Comment:
probably just one of the if condition is needed

Reply:
Add a task to check if we can refactor this code.